File: LivrusVirtus/VCPP6/wxWin/LibMySQL/ConProfissoes.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConProfissoes.h"

BEGIN_EVENT_TABLE(CfrmConProfissoes, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConProfissoes::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConProfissoes::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConProfissoes::OnBtnFechar)
  EVT_CLOSE(CfrmConProfissoes::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConProfissoes::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConProfissoes::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConProfissoes::CfrmConProfissoes(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConProfissoes::~CfrmConProfissoes() { }

void CfrmConProfissoes::IniciaControles() {
 this->SetTitle(STR_CONPROFISS);
 this->labDescricao->SetLabel(wxT("Profissão:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("Profissão"));
}

void CfrmConProfissoes::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_PROFISSOES);
  event.Skip();
}

void CfrmConProfissoes::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConProfissoes::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConProfissoes::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Profissao('S',wxT(""), edtDescricao->GetValue()), STR_PROFNENC, 2);
  event.Skip();
}

void CfrmConProfissoes::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConProfissoes::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

File: LivrusVirtus/BoaConstructor/wxPython/ConProfissoes.py
#Boa:Dialog:frmConProfissoes

from wxPython.wx import *
from wxPython.grid import *
from Consultas import frmConsultas
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource

def create(parent):
    return frmConProfissoes(parent)

class frmConProfissoes(frmConsultas):
    def _init_ctrls(self, prnt):
      frmConsultas._init_ctrls(self, prnt)
      self.SetTitle(Resource.STR_CONPROFISS)
      self.labDescricao.SetLabel("Profiss\xe3o: ")
      self.gridConsulta.SetColLabelValue(1, "Profiss\xe3o")

      EVT_BUTTON(self.btnPesquisar, self.btnPesquisar.GetId(), 
        self.OnBtnPesquisarClick)
      EVT_BUTTON(self.btnLimpar, self.btnLimpar.GetId(), self.OnBtnLimparClick)
      EVT_BUTTON(self.btnFechar, self.btnFechar.GetId(), self.OnBtnFecharClick)
      EVT_CLOSE(self, self.OnFrmConProfissoesClose)
      EVT_GRID_SELECT_CELL(self, self.OnGridSelectCell)
      EVT_GRID_LABEL_LEFT_CLICK(self, self.OnGridLabelLeftClick)

    def __init__(self, parent):
      self._init_ctrls(parent)

    def OnFrmConProfissoesClose(self, event):
      self.Fecha(Resource.MNU_CON_PROFISSOES)
      event.Skip()

    def OnBtnFecharClick(self, event):
      self.Fecha(0)
      event.Skip()

    def OnBtnLimparClick(self, event):
      self.LimparDados()
      event.Skip()

    def OnBtnPesquisarClick(self, event):
      self.PesquisarDados(
        ConsultasSQL.ConsSQL.Profissao('S', "", self.edtDescricao.GetValue()),
        Resource.STR_PROFNENC, 2)
      event.Skip()

    def OnGridSelectCell(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()

    def OnGridLabelLeftClick(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()